pkg/build/api/v1beta1/conversion.go

```diff
@@ -197,6 +221,18 @@ func init() {
 					out.Tag = tag
 				case "DockerImage":
 					out.Image = in.From.Name
+				case "ImageStream":
+					out.From = &kapi.ObjectReference{
+						Name:      in.From.Name,
+						Namespace: in.From.Namespace,
+						Kind:      in.From.Kind,
+					}
+				default:
```

Comment:
Sending any other kind is a validation error, not a conversion overwrite.  We should be rejecting unknown kinds in the core validation I think.